packages/event-bus-redis/src/services/event-bus-redis.ts

```diff
@@ -53,93 +56,71 @@ type EmitOptions = {
  * Can keep track of multiple subscribers to different events and run the
  * subscribers when events happen. Events will run asynchronously.
  */
-export default class EventBusService {
+export default class RedisEventBusService
+  extends TransactionBaseService
+  implements IEventBusService
+{
   protected readonly config_: ConfigModule
-  protected readonly manager_: EntityManager
   protected readonly logger_: Logger
-  protected readonly stagedJobRepository_: typeof StagedJobRepository
-  protected readonly jobSchedulerService_: JobSchedulerService
+  protected readonly stagedJobService_: StagedJobService
   protected readonly eventToSubscribersMap_: Map<
     string | symbol,
     SubscriberDescriptor[]
-  >
-  protected readonly redisClient_: Redis.Redis
-  protected readonly redisSubscriber_: Redis.Redis
-  protected queue_: Bull
+  > = new Map()
+
+  protected redisClient_: Redis
+  protected redisSubscriber_: Redis
+  protected queue_: Queue
   protected shouldEnqueuerRun: boolean
-  protected transactionManager_: EntityManager | undefined
   protected enqueue_: Promise<void>
 
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+
   constructor(
-    {
-      manager,
-      logger,
-      stagedJobRepository,
-      redisClient,
-      redisSubscriber,
-      jobSchedulerService,
-    }: InjectedDependencies,
+    container: MedusaContainer & InjectedDependencies,
     config: ConfigModule,
+    moduleDeclaration?: ConfigurableModuleDeclaration,
     singleton = true
   ) {
+    // @ts-ignore
+    super(...arguments)
+
+    if (moduleDeclaration?.resources !== MODULE_RESOURCE_TYPE.SHARED) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_ARGUMENT,
+        "At the moment this module can only be used with shared resources"
+      )
+    }
+
     this.config_ = config
-    this.manager_ = manager
-    this.logger_ = logger
-    this.jobSchedulerService_ = jobSchedulerService
-    this.stagedJobRepository_ = stagedJobRepository
-
-    if (singleton) {
-      const opts = {
-        createClient: (type: string): Redis.Redis => {
```

Comment:
self-review: `bullmq` ensures that we are reusing connections under the hood, so we no longer need this approach